onadata/libs/filters.py

```diff
@@ -67,24 +66,54 @@ def filter_queryset(self, request, queryset, view):
 
 class XFormPermissionFilterMixin(object):
 
+    @staticmethod
+    def _get_xform(request, queryset, view, keyword):
```

Comment:
Something seems to be going on here but I can't quite put my finger on it. I get success if I try something like `/metadata?xform_id=2&format=json`, but `404` if I try `/metadata/2.json`.

Reply:
There are 2 different urls.
In `/api/v2/metadata/2.json`, `2` is the ID of the metadata object. It's the detail endpoint.
While `/api/v2/metadata?xform_id=2&format=json` (or`/api/v2/metadata.json?xform_id=2`) is the the list endpoint filtered by XForm. 

The rules are:
- Users are allowed to see your metadata only (`/api/v2/metadata.json`) 
- Users are allowed to see metadata of publicly shared xforms when they access the link directly (`/api/v2/metadata?xform_id=2`)

Reply:
I see, so is your comment wrong on line 99 that says `if `xform` is provided (i.e.: `/api/v1/metadata.json?xform=1`) or they access a metadata object (i.e.: `/api/v1/metadata/1.json`)`?

Reply:
It's not wrong. If the user knows the direct link, they can access it. 
Direct link can be:
- The list endpoint : `/api/v1/metadata.json?xform=<xform_id>`
- The detail endpoint : `/api/v1/metadata/<metadata_id>.json`

I guess my comment needs more explanations